bubblegum/program/tests/simple.rs

```diff
@@ -54,54 +65,211 @@ async fn test_create_tree_and_mint_passes() {
+        // of the ME creation invoked within decompress.
+        assert_eq!(
+            mint.mint_authority,
+            COption::Some(find_master_edition_account(&mint_key).0)
+        );
+        assert_eq!(mint.supply, 1);
+        assert_eq!(mint.decimals, 0);
+
+        assert!(mint.is_initialized);
+        assert!(mint.freeze_authority.is_none());
+
+        let token_account_key = get_associated_token_address(&leaf.owner.pubkey(), &mint_key);
+        let token_account = tree.read_account(token_account_key).await.unwrap();
+        let t = spl_token::state::Account::unpack(token_account.data.as_slice()).unwrap();
+
+        assert_eq!(t.mint, mint_key);
+        assert_eq!(t.owner, leaf.owner.pubkey());
+        assert_eq!(t.amount, 1);
+        assert_eq!(t.state, spl_token::state::AccountState::Initialized);
+        assert_eq!(t.delegated_amount, 0);
+
+        assert!(t.delegate.is_none());
+        assert!(t.is_native.is_none());
+        assert!(t.close_authority.is_none());
+
+        let metadata_key = find_metadata_account(&mint_key).0;
+        let meta_account = tree.read_account(metadata_key).await.unwrap();
+        let meta: mpl_token_metadata::state::Metadata =
+            mpl_token_metadata::state::Metadata::safe_deserialize(meta_account.data.as_slice())
+                .unwrap();
+
+        assert_eq!(meta.key, mpl_token_metadata::state::Key::MetadataV1);
+        // assert_eq!(meta.update_authority, ?);
+        assert_eq!(meta.mint, mint_key);
+        assert_eq!(
+            meta.data.name,
+            puffed_out_string(&leaf.metadata.name, MAX_NAME_LENGTH)
+        );
+        assert_eq!(
+            meta.data.symbol,
+            puffed_out_string(&leaf.metadata.symbol, MAX_SYMBOL_LENGTH)
+        );
+        assert_eq!(
+            meta.data.uri,
+            puffed_out_string(&leaf.metadata.uri, MAX_URI_LENGTH)
+        );
+        assert_eq!(
+            meta.data.seller_fee_basis_points,
+            leaf.metadata.seller_fee_basis_points
+        );
+
+        let _meta_creators = meta.data.creators.as_ref().unwrap();
+
+        // TODO: It appears that an extra creator (with share `0` and verified) appears in the
+        // metadata post decompress. Need to investigate more here.
+
+        // assert_eq!(meta_creators.len(), leaf.metadata.creators.len());
+        // // Can't compare directly as they are different types for some reason.
+        // for (i, c1) in meta_creators.iter().enumerate() {
+        //     let c2 = leaf.metadata.creators[i];
+        //     assert_eq!(c1.address, c2.address);
+        //     assert_eq!(c1.verified, c2.verified);
+        //     assert_eq!(c1.share, c2.share);
+        // }
+
+        assert!(!meta.primary_sale_happened);
+        assert!(!meta.is_mutable);
+
+        // TODO: Is the edition nonce field predictable so we can/should test for a specific
+        // value there as well?
```

Comment:
I can look at this tomorrow or you can ask @samuelvanderwaal he would probably know off the top of his head.